File: app/src/main/java/com/sagalogistics/backend/database/OrderDAO.java
package com.sagalogistics.backend.database;

import com.sagalogistics.backend.models.Order;

import java.util.concurrent.Future;

public interface OrderDAO {
    void add(Order order);
    Future<Order> get(String key);
    void update(String key, Order order);
    void delete(String key);
    void deleteItem(String orderKey, String itemKey);
}

File: app/src/main/java/com/sagalogistics/backend/database/OrderDAO.kt
package com.sagalogistics.backend.database

import com.sagalogistics.backend.models.Order
import java.util.concurrent.Future

interface OrderDAO {
    fun add(order: Order)
    fun get(key: String): Future<Order?>
    fun update(key: String, order: Order)
    fun delete(key: String)
    fun deleteItem(itemKey: String)
}